src/vmcs/src/ept_intel_x64.cpp

```diff
@@ -41,60 +41,66 @@ ept_intel_x64::ept_intel_x64(pointer epte)
 ept_entry_intel_x64
 ept_intel_x64::add_page(integer_pointer gpa, integer_pointer bits, integer_pointer end)
 {
-    auto &&index = ept::index(gpa, bits);
+    auto index = ept::index(gpa, bits);
+    auto view = gsl::make_span(m_ept, ept::num_entries);
```

Comment:
please move this to a new constructor for the entry class. Basically, you would just give it the ept pointer, and an index, and it will grab a reference to the entry on it's own using the span